src/common/snippets/include/snippets/utils/reg_utils.hpp

```diff
@@ -16,9 +16,9 @@ inline static std::vector<size_t> transform_snippets_regs_to_idxs(const std::vec
     std::vector<size_t> idxs;
     idxs.reserve(regs.size());
     for (const auto& reg : regs) {
-        OPENVINO_ASSERT(reg.type == snippets::RegType::address || expected_type == snippets::RegType::undefined ||
-                            reg.type == expected_type,
-                        "Reg type mismatch during to_idxs conversion");
+        bool is_type_valid = (expected_type == snippets::RegType::undefined) ||
```

Comment:
```suggestion
        const auto is_type_valid = (expected_type == snippets::RegType::undefined) ||
```
If this flag is read-only to be more const correct. 
Check all similar modifications, please.